### PR TITLE
[NET-690] Hide Types in sidebar/dropdowns if user has no acl permissions

### PR DESCRIPTION
**Pull request by doblanken**
[NET-690](https://nethinks.atlassian.net/browse/NET-690)

### CODE REVIEW

File: app/src/app/framework/services/type.service.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ import {
 } from '../../services/models/api-response';
 import { CollectionParameters } from '../../services/models/api-parameter';
 import { ValidatorService } from '../../services/validator.service';
+import {UserService} from '../../management/services/user.service';
```
Comment (johndgrey) [inline app/src/app/framework/services/type.service.ts:35]: Please look for spaces. Possibly the tslint.json must be adapted.

**Review by johndgrey — APPROVED** [review]

File: app/src/app/framework/services/type.service.ts [hunk 2]
```diff
@@ -62,7 +64,40 @@ export class TypeService<T = CmdbType> implements ApiService {
...
-  constructor(private api: ApiCallService) {
+  constructor(private api: ApiCallService, private userService: UserService) {
+  }
+
+  /**
+   * Returns acl read filter
+   *
+   * @private
+   */
+  private getAclReadFilter() {
```
Comment (markheumueller) [inline app/src/app/framework/services/type.service.ts:75]: Why do you out-stock this function? If I see correctly it is only used in one place.

File: app/src/app/framework/services/type.service.ts [hunk 3]
```diff
@@ -62,7 +64,40 @@ export class TypeService<T = CmdbType> implements ApiService {
...
+  }
+
+  /**
+   * Filters all types which don't include the READ right
+   *
+   * @param types @link{CmdbType[]}
+   * @param right
+   * @return @link{CmdbType[]}
+   */
+  public filterTypesByAcl(types: CmdbType[], right: AccessControlPermission): CmdbType[] {
```
Comment (markheumueller) [inline app/src/app/framework/services/type.service.ts:97]: Ok, this is not a function that belongs in the type service. It is used for a specific selection of components. Either it is used on the component level itself or better yet, the data does not get into the frontend at all.

File: app/src/app/framework/category/category-form/category-form.component.ts [hunk 4]
```diff
@@ -140,7 +141,7 @@ export class CategoryFormComponent implements OnInit, OnChanges, OnDestroy {
       for (const type of this.$category.types) {
         buffer.push(this.findAssignedTypeByIndex(type));
       }
-      this.assignedTypes = buffer;
+      this.assignedTypes = this.typeService.filterTypesByAcl(buffer, AccessControlPermission.READ);
```
Comment (markheumueller) [inline app/src/app/framework/category/category-form/category-form.component.ts:144]: The filter should be applied on this level.

**Review by markheumueller — CHANGES_REQUESTED** [review]

**Comment by markheumueller** [comment]
But the rest of it is looking good. Refactor only the filter usage.

File: app/src/app/framework/services/type.service.ts (hunk 2 above)
Comment (doblanken) [reply app/src/app/framework/services/type.service.ts:75]: This is used at 4 different places, getTypeList, getTypesByNameOrLabel, getUncategorizedTypes and getTypeListByCategory

**Review by johndgrey — APPROVED** [review]

**Pull request merged**